syntax = "proto3";

package reservation;

service HotelReservation {
  rpc ReserveHotel (HotelRequest) returns (HotelResponse);
}

service FlightReservation {
  rpc ReserveFlight (FlightRequest) returns (FlightResponse);
}

message HotelRequest {
  string hotel_id = 1;
  string user_id = 2;
  string check_in_date = 3;
  string check_out_date = 4;
  int32 number_of_rooms = 5;
}

message HotelResponse {
  string reservation_id = 1;
  bool success = 2;
  string message = 3;
}

message FlightRequest {
  string flight_id = 1;
  string user_id = 2;
  string departure_date = 3;
  string return_date = 4;
  int32 number_of_tickets = 5;
}

message FlightResponse {
  string reservation_id = 1;
  bool success = 2;
  string message = 3;
}